pages/tidal tools/sourcecode.md

```diff
@@ -44,32 +44,92 @@ An example of some of the criteria that are considered include:
 
-Once it is complete you can view your application and the newly updated data in Tidal Migrations, it can take a couple minutes for the data to be processed and uploaded.
 
-Try it out!
+## Perform the analysis
+
+When analyzing a source code (or multiple), You have two options. 
+
+1. Perform the source code analysis and upload the result immediately to your workspace. This is Tidal Tools' default behaviour. To do so, all you need to do is run the following command.
+
+    ```bash
+    cd /path/to/source-code
+    tidal analyze code --app-id 111
+    ```
+
+    Alternatively, you can pass the location to the source code.
+    ```bash
+    tidal analyze code ~/location/to-source-code/ --app-id 111
+    ```
+
+2. Perform the source code analysis and upload the results **at a later time**. The following section will explain how to run database analysis in **offline mode**
+
+### Running offline
+
+There are circumstances in which you need to perform a source code analysis on an environment without or with restricted internet access. In such case, You can perform the analysis, capture its results and at a later stage upload those results to your Tidal Migrations workspace.
+
+These are the steps you need to follow in order to bypass internet access limitations:
+
+1. On a Machine with internet access, you need to install, and configure Tidal Tools. 
+
+2. Package Tidal Tools and its required docker container images into a tar file, This will allow you to move the archive file into your restricted environment. To do so, run the following command.
+
+    `tidal backup`
+
+    Once it has finished, you will find (in your current location) a tar file called `tidal-snapshot_DATE.tar`. This is the file you need to transport into your internet restricted environment.
+
+3. On the machine that has no internet access. You will now restore Tidal Tools using the following command.
+
+    `tidal restore tidal-snapshot_DATE.tar`
+
+    This will load a docker image and all the existing Tidal Tools configurations from the original machine. You can now run the source code analysis without any external network connectivity.:
+
+    `tidal analyze code [path/to-directory] --offline --output-dir [directory]`
+
+    Note:
+    -  The `--offline` flag, it will indicate to Tidal Tools that the output needs to be stored in a file instead of being uploaded.
+    - The `--output-dir`, will be use to determine where the output will be located.
+
+    After the analysis is completed, you will find a zip file called, `result-<DATE>.zip` that can then be transferred into a machine with internet connectivity.
+
+
+4. Back to the machine with internet access. you can now upload your results to your workspace with this command.
+
+    `tidal analyze code --app-id [app_id_for_your_application] --upload [file_name]`
+
+    You should receive confirmation that the upload has been completed and can navigate to Tidal Migrations to see the results.
```

Comment:
I wasn't aware of this. How do they get the confirmation? Email?

Reply:
Yeah, that is right. You will get an email.